storage/aptosdb/src/backup/backup_data_accessor.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{backup::backup_handler::DbState, fast_sync_aptos_db::FastSyncStorageWrapper};
+use anyhow::Result;
+use aptos_crypto::hash::HashValue;
+use aptos_types::{
+    contract_event::ContractEvent,
+    ledger_info::LedgerInfoWithSignatures,
+    proof::{SparseMerkleRangeProof, TransactionAccumulatorRangeProof, TransactionInfoWithProof},
+    state_store::{state_key::StateKey, state_value::StateValue},
+    transaction::{Transaction, TransactionInfo, Version},
+    write_set::WriteSet,
+};
+use std::sync::Arc;
+
+/// `BackupDataAccessor` provides functionalities for data backup.
+///  This is supposed to be replace backup handler.
+#[derive(Clone)]
+pub struct BackupDataAccessor {
```

Comment:
Silly question: why do we need this? If all calls go directly to the underlying wrapper DB, can we not just pass the wrapper to whatever objects require this?

I likely lack context here though 😄

Reply:
yes, we can directly use the DBWrapper for Backup Service to avoid introducing backup_data_acessor. I was just following the convention of BackupHandler